vmware_exporter/helpers.py

```diff
@@ -8,30 +8,33 @@ def batch_fetch_properties(content, obj_type, properties):
         recursive=True
     )
 
-    PropertyCollector = vmodl.query.PropertyCollector
-
-    # Describe the list of properties we want to fetch for obj_type
-    property_spec = PropertyCollector.PropertySpec()
-    property_spec.type = obj_type
-    property_spec.pathSet = properties
-
-    # Describe where we want to look for obj_type
-    traversal_spec = PropertyCollector.TraversalSpec()
-    traversal_spec.name = 'traverseEntities'
-    traversal_spec.path = 'view'
-    traversal_spec.skip = False
-    traversal_spec.type = view_ref.__class__
-
-    obj_spec = PropertyCollector.ObjectSpec()
-    obj_spec.obj = view_ref
-    obj_spec.skip = True
-    obj_spec.selectSet = [traversal_spec]
-
-    filter_spec = PropertyCollector.FilterSpec()
-    filter_spec.objectSet = [obj_spec]
-    filter_spec.propSet = [property_spec]
-
-    props = content.propertyCollector.RetrieveContents([filter_spec])
+    try:
+        PropertyCollector = vmodl.query.PropertyCollector
+
+        # Describe the list of properties we want to fetch for obj_type
+        property_spec = PropertyCollector.PropertySpec()
+        property_spec.type = obj_type
+        property_spec.pathSet = properties
```

Comment:
Should we consider grabbing all properties and then use a blacklist approach further down?

Reply:
We could try this later, but right now that would mean we fetched a lot more data over the wire and then didn't do anything with it. So it would probably be slower (but I haven't timed it) and either way it would get us any more useful data in the exporter. So I don't think this is a change to do in this PR.